src/openforms/registrations/contrib/objects_api/tests/test_utils.py

```diff
@@ -63,3 +69,46 @@ def test_different_input(self):
                 result = html_escape_json(sample)
 
                 self.assertIs(result, sample)
+
+    def test_render_to_json_large_numbers(self):
+        payment = SubmissionPaymentFactory.create(
+            amount=1_000,
+        )
+
+        context_data = get_payment_context_data(payment.submission)
+
+        self.assertEqual("1000.00", context_data["amount"])
+
+    def test_render_to_json_many_decimal_places(self):
+        payment = SubmissionPaymentFactory.create(
+            amount=3.1415926535,
+        )
+
+        context_data = get_payment_context_data(payment.submission)
+
+        self.assertEqual("3.14", context_data["amount"])
+
+    def test_multiple_public_order_ids(self):
+        submission = SubmissionFactory.create()
+        payment1, payment2 = SubmissionPaymentFactory.create_batch(
+            2, submission=submission, status=PaymentStatus.completed, amount=10
+        )
+
+        template = textwrap.dedent(
+            """{
+            "completed": {% if payment.completed %}true{% else %}false{% endif %},
+            "amount": {{ payment.amount }},
+            "public_order_ids": {{ payment.public_order_ids }}
```

Comment:
I don't understand how this variable results in valid JSON other than pure coincidence. In fact, it may not even be valid JSON due to python strings typically being printend with single quotes rather than double quotes.